aiocogeo/tiler.py

```diff
@@ -183,51 +222,69 @@ async def tile(
         tile_y: int,
         tile_z: int,
         tilesize: int = 256,
-        resample_method: int = Image.NEAREST,
-        tms: TileMatrixSet = DEFAULT_TMS,
+        indexes: Optional[Union[int, Sequence]] = None,
+        expression: Optional[str] = "",  # placeholder
+        resampling_method: str = "nearest",
     ) -> TileResponse:
+
+        if isinstance(indexes, int):
+            indexes = (indexes,)
+
         tile = morecantile.Tile(x=tile_x, y=tile_y, z=tile_z)
-        tile_bounds = tms.xy_bounds(tile)
+        tile_bounds = self.tms.xy_bounds(*tile)
+
+        # resampling_method -> resample_method
+        resample_method = Resampling[resampling_method]
+
         width = height = tilesize
-        if self.cog.epsg != tms.crs:
+        if self.epsg != self.tms.crs.to_epsg():
             arr = await self._warped_read(
                 tile_bounds,
                 width,
                 height,
-                bounds_crs=tms.crs,
+                bounds_crs=self.tms.crs,
                 resample_method=resample_method,
             )
         else:
-            arr = await self.cog.read(
+            arr = await self.read(
                 tile_bounds, shape=(width, height), resample_method=resample_method
             )
 
-        return TileResponse.from_array(arr)
+        # should be handled at read level
```

Comment:
👀

Reply:
oh yes that would be a really nice optimization